app/main/views/frameworks.py

```diff
@@ -207,3 +216,22 @@ def _framework_updates_page(error_message=None):
         error_message=error_message,
         **template_data
     ), status_code
+
+
+def _get_declaration_status():
+    try:
+        answers = data_api_client.get_selection_answers(
+            current_user.supplier_id, 'g-cloud-7'
+        )['selectionAnswers']['questionAnswers']
+    except APIError as e:
+        if e.status_code == 404:
+            return 'unstarted'
+        else:
+            abort(e.status_code)
+
+    if not answers:
+        return 'unstarted'
+    elif all(keys in answers for keys in ONE_REQUIRED_QUESTION_FROM_EACH_PAGE):
```

Comment:
I think we should validate the full declaration to determine the status, not check a subset of questions that we assume cover all the pages. And we should store the status once the declaration is complete since "declaration is complete" is used by the API MI stats.